librewolf 1.36.0.1,1

creativecoder opened the pull request:
Currently the `autobump` workflow is not finding the latest version of LibreWolf because of how the versioning scheme works.

- Current version: `136.0.1-1`
- Latest brew version: `136.0-2`

The `librewolf` cask version combines a Firefox version with a LibreWolf release number separated by `-`, e.g. `136.0.1-1`. However, when the Firefox version has only 2 places, the LibreWolf release is treated as the final part of the Firefox version.

So `[136, 0, 2]` is considered newer than `[136, 0, 1, 1]`.

This change to the `librewolf` formula updates the separator  from `-` to `,`, e.g. `136.0-2` becomes `136.0,2` to make sure the version comparison works correctly for detecting updates.

Output of `brew livecheck --cask --debug --autobump librewolf`:

**Before**
```
Cask:             librewolf
livecheck block?: Yes

URL:              https://gitlab.com/librewolf-community/browser/bsys6.git
Strategy:         Git

Matched Versions:
111.0-2, 111.0-2-2, 111.0-3, 111.0.1-1, 112.0-1, 112.0.1-2, 112.0.2-1, 113.0-3, 113.0.1-1, 113.0.2-1, 114.0-1, 114.0.1-2, 114.0.2-1, 115.0.1-1, 115.0.2-2, 116.0-1, 116.0.2-1, 116.0.3-1, 117.0-1, 117.0.1-1, 118.0.1-1, 118.0.2-1, 119.0-4, 119.0-5, 119.0-6, 119.0-7, 119.0.1-1, 120.0-1, 120.0-2, 120.0.1-1, 121.0-1, 122.0-1, 122.0-2, 122.0.1-2, 123.0-1, 123.0.1-1, 124.0.1-1, 125.0.2-1, 125.0.3-1, 126.0-1, 126.0.1-1, 127.0-1, 127.0.1-1, 127.0.2-2, 128.0-2, 128.0.3-1, 128.0.3-2, 129.0-1, 129.0.1-1, 129.0.2-1, 130.0-1, 130.0-2, 130.0-3, 130.0.1-1, 131.0-1, 131.0.2-1, 131.0.3-1, 132.0-1, 132.0.1-1, 132.0.2-1, 133.0-1, 133.0-3, 133.0.3-1, 134.0-1, 134.0.1-1, 134.0.2-1, 135.0-1, 135.0.1-1, 136.0-1, 136.0-2, 136.0.1-1

librewolf: 136.0-2 ==> 136.0-2
```

**After**
```
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromNameLoader): loading librewolf

Cask:             librewolf
livecheck block?: Yes

URL:              https://gitlab.com/librewolf-community/browser/bsys6.git
Strategy:         Git
Regex:            /^v?(\d+(?:[.-]\d+)+)$/i

Matched Versions:
111.0,2, 111.0,2,2, 111.0,3, 111.0.1,1, 112.0,1, 112.0.1,2, 112.0.2,1, 113.0,3, 113.0.1,1, 113.0.2,1, 114.0,1, 114.0.1,2, 114.0.2,1, 115.0.1,1, 115.0.2,2, 116.0,1, 116.0.2,1, 116.0.3,1, 117.0,1, 117.0.1,1, 118.0.1,1, 118.0.2,1, 119.0,4, 119.0,5, 119.0,6, 119.0,7, 119.0.1,1, 120.0,1, 120.0,2, 120.0.1,1, 121.0,1, 122.0,1, 122.0,2, 122.0.1,2, 123.0,1, 123.0.1,1, 124.0.1,1, 125.0.2,1, 125.0.3,1, 126.0,1, 126.0.1,1, 127.0,1, 127.0.1,1, 127.0.2,2, 128.0,2, 128.0.3,1, 128.0.3,2, 129.0,1, 129.0.1,1, 129.0.2,1, 130.0,1, 130.0,2, 130.0,3, 130.0.1,1, 131.0,1, 131.0.2,1, 131.0.3,1, 132.0,1, 132.0.1,1, 132.0.2,1, 133.0,1, 133.0,3, 133.0.3,1, 134.0,1, 134.0.1,1, 134.0.2,1, 135.0,1, 135.0.1,1, 136.0,1, 136.0,2, 136.0.1,1

librewolf: 136.0,2 ==> 136.0.1,1
```

---

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

khipp commented:
Thank you, @creativecoder! I think the underlying issue is best resolved in the `brew` repository rather than by working around it in a cask. There may be other casks or formulae where the current version comparison could lead to incorrect results.

CC @samford

creativecoder commented:
Thanks for taking a look @khipp ! Some questions:

First, would it still be worth moving forward with this PR in the meantime, so that brew cask users can update LibreWolf to the latest version?

> I think the underlying issue is best resolved in the brew repository rather than by working around it in a cask.

Should we discuss this in a new brew issue, then? Any suggestions for implementation? Seems like it might be tricky to maintain backwards compatibility with current APIs, but I haven't looked very far into it.

samford commented:
Off the top of my head, I don't think it will be easy to solve this in brew (but I would like to be wrong). The fundamental issue is that `Version` doesn't distinguish between delimiters like `.` and `-`. With a version like `1.2.3-4`, it may make sense to compare the parts before and after the hyphen separately but implementing that could prove challenging. I don't have the bandwidth right now but if anyone wants to dig into it, the [`Version`](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/version.rb) class is the place to look.

khipp commented:
> Should we discuss this in a new brew issue, then? Any suggestions for implementation? Seems like it might be tricky to maintain backwards compatibility with current APIs, but I haven't looked very far into it.

It would be great to open a corresponding issue in the `brew` repository for tracking so that someone can dig into it if the bandwidth is available.

> First, would it still be worth moving forward with this PR in the meantime, so that brew cask users can update LibreWolf to the latest version?

That makes sense to me. I added a commit that simplifies the `livecheck` slightly by splitting the Git tags on `-` delimiters. The resulting parts will be compared as units, ensuring that the version component delimited by `.` takes precedence.

creativecoder commented:
> I added a commit that simplifies the livecheck slightly by splitting the Git tags on - delimiters. The resulting parts will be compared as units, ensuring that the version component delimited by . takes precedence.

Looks good to me. I updated the PR description to match the changes to the code. Thank you!
